galley/pkg/config/analysis/diag/message.go

```diff
@@ -76,6 +77,17 @@ func (m *Message) String() string {
 		"%v [%v]%s %s", m.Type.Level(), m.Type.Code(), origin, fmt.Sprintf(m.Type.Template(), m.Parameters...))
 }
 
+// MarshalJSON satisfies the Marshaler interface
```

Comment:
add `json` tags to the struct and let json marshaler do the rest?

Reply:
I thought about that, but right now there are unexported fields and nested stuff that I'd have to change. After the formatting PR lands I'm planning on putting work into the structure so I'd like to beg off this issue until then. If you feel strongly it's worth doing now lmk and I'll move stuff around.

Reply:
Not super urgent. I think we will end up having the message be an API and there we can utilize the native json marshaling.